src/main/java/ConsoleInput.java

```diff
@@ -1,91 +1,96 @@
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.io.InputStreamReader;
+import java.io.*;
 import java.time.LocalDate;
 import java.time.format.DateTimeFormatter;
 import java.util.regex.Pattern;
 
 public class ConsoleInput {
 
-    public ConsoleInput() {
+    private final InputStream input;
+    private final OutputStream output;
+    private final BufferedReader reader;
+    private final PrintStream printer;
+
+    public ConsoleInput(InputStream input, OutputStream output) {
+        reader = new BufferedReader(new InputStreamReader(input));
+        printer = new PrintStream(output);
+        this.input = input;
+        this.output = output;
     }
 
-    public String confirmInput(String detail) {
+    public String confirmInput(String field, boolean update) {
         Boolean validInput = false;
         String userInput = null;
         while (!validInput) {
-            userInput = getInput(detail);
-            validInput = validateInput(detail, userInput);
+            userInput = getInput(field);
+            validInput = validateInput(field, userInput, update);
         }
         return userInput;
     }
 
-    public int contactChoice() {
-        return Integer.parseInt(confirmInput("contact choice"));
-    }
-
     public int menuChoice() {
-        String input;
+        String userInput;
         try {
-            BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
-            input = reader.readLine();
+            userInput = reader.readLine();
         } catch (IOException e) {
             return - 1;
         }
         try {
-            return Integer.parseInt(input);
+            return Integer.parseInt(userInput);
         } catch (NumberFormatException e) {
             return 0;
         }
     }
 
     public String getInput(String detail) {
-        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
-        System.out.println("Please enter your " + detail + ":");
-        String input = null;
+        printer.println("Please enter your " + detail + ":");
+        String userInput = null;
         try {
-            input = reader.readLine();
+            userInput = reader.readLine();
         } catch (IOException e) {
-            System.out.println("Cannot read line" + e);
+            printer.println("Cannot read line" + e);
         }
-        return input;
+        return userInput;
     }
 
-    public Boolean validateInput(String detail, String input) {
+    public Boolean validateInput(String detail, String userInput, boolean update) {
```

Comment:
Feels like this should be a responsibility of the Contact to maintain understanding of how to validate it's own fields.